task-sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -1147,8 +1151,22 @@ def final_state(self):
             return self._terminal_state or TaskInstanceState.SUCCESS
         if self._exit_code != 0 and self._terminal_state == SERVER_TERMINATED:
             return SERVER_TERMINATED
+
+        if self._exit_code != 0 and self._is_signal_retryable() and self._should_retry:
+            return TaskInstanceState.UP_FOR_RETRY
+
         return TaskInstanceState.FAILED
 
+    def _is_signal_retryable(self) -> bool:
+        """Check if the exit code signal can be retried."""
+        if self._exit_code is None:
+            return False
+
+        if self._exit_code == -signal.SIGKILL or self._exit_code == -signal.SIGTERM:
+            return True
```

Comment:
Why only sigkill and sigterm? I think any signal that kills the task should be up for a retry -- SIGSEGV (seg fault) for instance could just as likely be retried.